### PR TITLE
Feature: Performance overview table with indicator linked

### PR DESCRIPTION
**Pull request by delemaf**
### :pushpin: References

-   **Issue:** [Closes](https://app.clickup.com/t/8694tp0tp) #8694tp0tp

### :memo: Implementation
Needs https://github.com/EyeSeeTea/d2-api/pull/150 to be merged before (it fixes ts error)
- Dashboard page
- Add stats cards for Respond, alert, watch
- Add performance overview table
- Date Range Filter
- Filtering

### :video_camera: Screenshots/Screen capture
image
image

### CODE REVIEW

File: src/data/repositories/AnalyticsD2Repository.ts [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+}
+
+interface HazardEntry {
+    hazard: string;
+    total: number;
+}
+
+export type DiseaseTotalAttrs = DiseaseEntry | HazardEntry;
+
+export class AnalyticsD2Repository implements AnalyticsRepository {
```
Comment (9sneha-n) [inline src/data/repositories/AnalyticsD2Repository.ts:53]: analytics, indicators etc are DHIS concepts. As per clean architecture the repository should only deal with domain entities. So you can you rename the class, file and functions to not have any references to DHIS concepts but instead refer to Zebra concepts - performance overview table, performance overview repository etc.

File: src/data/repositories/common/getProgramIndicatorsFromDatastore.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import { Maybe } from "../../../utils/ts-utils";
+import { FutureData } from "../../api-futures";
+import { DataStoreClient } from "../../DataStoreClient";
+
+export enum ProgramIndicatorsDatastoreKey {
```
Comment (9sneha-n) [inline src/data/repositories/common/getProgramIndicatorsFromDatastore.ts:5]: again, the enums and types should not refer to DHIS concepts like  ProgramIndicators, please rename.

File: src/data/repositories/consts/AnalyticsConstants.ts [hunk 3]
```diff
@@ -0,0 +1,221 @@
+export enum IndicatorsId {
+    suspectedDisease = "jLvbkuvPdZ6",
```
Comment (9sneha-n) [inline src/data/repositories/consts/AnalyticsConstants.ts:2]: I am guessing that these will be fetched from datastore in upcoming PRs?

File: src/webapp/components/date-picker/DateRangePicker.tsx [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+            if (!value || value.length !== 2) {
+                setStartDate(moment().startOf("month").toDate());
+                setEndDate(moment().toDate());
+            }
+        }, [value]);
+
+        // Adjust startDate if endDate < startDate
+        useEffect(() => {
+            if (endDate && startDate && moment(endDate).isBefore(startDate)) {
+                setStartDate(endDate);
```
Comment (9sneha-n) [inline src/webapp/components/date-picker/DateRangePicker.tsx:34]: nice!

File: src/webapp/components/table/statistic-table/StatisticTable.tsx [hunk 5]
```diff
@@ -67,6 +76,21 @@ export const StatisticTable: React.FC<StatisticTableProps> = React.memo(
             columnRules
         );
 
+        const onOrderBy = useCallback(
+            (value: string) =>
+                setOrder &&
+                setOrder({
+                    name: value,
```
Comment (9sneha-n) [inline src/webapp/components/table/statistic-table/StatisticTable.tsx:83]: whenever your set of a react state depends on its previous state, you need to use an updater function. https://react.dev/reference/react/useState#updating-state-based-on-the-previous-state.

File: src/webapp/pages/dashboard/usePerformanceOverview.ts [hunk 6]
```diff
@@ -1,52 +1,62 @@
...
-        PerformanceOverviewData[]
+        ProgramIndicatorBaseAttrs[]
     >([]);
     const [isLoading, setIsLoading] = useState(false);
+    const [order, setOrder] = useState<Order>();
+
+    useEffect(() => {
+        if (dataPerformanceOverview) {
+            setDataPerformanceOverview(newDataPerformanceOverview =>
+                _(newDataPerformanceOverview)
```
Comment (9sneha-n) [inline src/webapp/pages/dashboard/usePerformanceOverview.ts:33]: whenever your set of a react state depends on its previous state, you need to use an updater function. https://react.dev/reference/react/useState#updating-state-based-on-the-previous-state. Then add dataPerformanceOverviewdata as a dependency to the useEffect

**Review by 9sneha-n — CHANGES_REQUESTED** [review]
thanks @delemaf Some changes requested

File: src/webapp/pages/dashboard/useDiseasesTotal.ts [hunk 7]
```diff
@@ -0,0 +1,37 @@
+import { useEffect, useState } from "react";
+import { useAppContext } from "../../contexts/app-context";
+import _ from "../../../domain/entities/generic/Collection";
+import { DiseaseTotalAttrs } from "../../../data/repositories/AnalyticsD2Repository";
+
+type State = {
+    diseasesTotal: any[];
```
Comment (9sneha-n) [inline src/webapp/pages/dashboard/useDiseasesTotal.ts:7]: diseasesTotal should be type safe, avoid using any

File: src/webapp/pages/dashboard/usePerformanceOverview.ts [hunk 8]
```diff
@@ -1,52 +1,62 @@
...
+import _ from "../../../domain/entities/generic/Collection";
 
-import { FilterType, TableColumn } from "../../components/table/statistic-table/StatisticTable";
-import { Id } from "../../../domain/entities/Ref";
+import { FiltersConfig, TableColumn } from "../../components/table/statistic-table/StatisticTable";
+import { ProgramIndicatorBaseAttrs } from "../../../data/repositories/AnalyticsD2Repository";
 
 type State = {
     columns: TableColumn[];
     dataPerformanceOverview: any[];
```
Comment (9sneha-n) [inline src/webapp/pages/dashboard/usePerformanceOverview.ts:10]: dataPerformanceOverview should be type safe, avoid using any

File: src/data/repositories/consts/AnalyticsConstants.ts (hunk 3 above)
Comment (delemaf) [reply src/data/repositories/consts/AnalyticsConstants.ts:2]: Yes!

**Review by 9sneha-n — APPROVED** [review]

**Pull request merged**